crypto/pkcs7/pk7_doit.c

```diff
@@ -316,17 +316,12 @@ BIO *PKCS7_dataInit(PKCS7 *p7, BIO *bio)
         fetched_cipher = EVP_CIPHER_fetch(p7_ctx->libctx,
                                           EVP_CIPHER_name(evp_cipher),
                                           p7_ctx->propq);
+        (void)ERR_pop_to_mark();
```

Comment:
cipher can not possibly be NULL in the old test since this entire if is nested inside a if (evp_cipher != NULL)